src/bp/nlu-core/intents/base-intent-classifier.ts

```diff
@@ -0,0 +1,135 @@
+import { MLToolkit } from 'botpress/sdk'
+import _ from 'lodash'
+import { NONE_INTENT } from 'nlu-core/training-pipeline'
+import { ListEntityModel, PatternEntity, Tools } from 'nlu-core/typings'
+import Utterance from 'nlu-core/utterance/utterance'
+
+import { BuildExactMatchIndex, ExactMatchIndex, findExactIntent } from './exact-matcher'
+import { IntentClassifier, IntentPredictions, IntentTrainInput } from './intent-classifier'
+import { getIntentFeatures } from './intent-featurizer'
+
+export const MIN_NB_UTTERANCES = 3
+
+export interface IntentClassifierModel {
+  svmModel: string | undefined
+  exact_match_index: ExactMatchIndex
+  list_entities: ListEntityModel[]
+  pattern_entities: PatternEntity[]
+}
+
+export class BaseIntentClassifier implements IntentClassifier {
```

Comment:
Why Base ? does that mean that you'll inherit from it ? kindof confusing

Reply:
From your typings this seems to be a NoneableIntentClassifier since it can predict None ? 

It's a bit confusing to be honest why this Noneable type ? It's just a text classifier in which we append a none class ... but it's still a Noneable ? 

Or is it that Noneable is our "out of scope" ?

Reply:
Youre right, naming is not quite right.

In the next incoming PR, I fully removed the notion of `None` intent in this classifier. So this one will become a true `unaware of the exitance of none` classifier.

Will be contained and used by a "NoneableClassifier".